### PR TITLE
Database Pagination

### LINKED ISSUE #23: Handle pagination / listing all results from the DB
This call may simply return a subset of all the results. Might need to do some cursor tracking

_Originally posted by @andresuribe87 in https://github.com/TBD54566975/did-dht-method/pull/21#discussion_r1367580173_

### PR DESCRIPTION
**Pull request by finn-block**
This adds DB pagination, addressing #23. The text of #23 suggests using a cursor, but that would prevent us from having pagination at the HTTP request level (that is, an HTTP request could not return a "page" of results, as the cursor would get dropped across requests), so I'm using an offset + page size approach.

How do we feel about using something like sqlite for local dev work? That will allow us to have queries and patterns that much more closely replicate the production database.

### CODE REVIEW

File: impl/pkg/storage/pkarr/pkarr.go [hunk 1]
```diff
@@ -9,3 +14,8 @@ type Record struct {
 	Sig []byte `json:"sig" validate:"required"`
 	Seq int64  `json:"seq" validate:"required"`
 }
+
+func (r Record) String() string {
+	encoding := base64.URLEncoding
```
Comment (thereisnogabe) [inline impl/pkg/storage/pkarr/pkarr.go]: should we use raw encoding?

**Review by thereisnogabe — APPROVED** [review]
nice work!

**Comment by thereisnogabe** [comment]
@finn-tbd does your current approach negate the need to replace bolt with sqlite, or is that something you would still like to explore?

**Comment by finn-block** [comment]
I've negated that need for now, but I definitely think having similar databases in dev and prod would be ideal. The behavior between boltdb and postgres is a little different even in this implementation: newly added records go to the end of the "all records" list in postgres, but in boltdb the sorting seems to be somewhat different (i haven't figured out specifics)

**Comment by thereisnogabe** [comment]
agree with that reasoning. please feel free to open up an issue to track

**Pull request merged**